users/tests.py

```diff
@@ -1,3 +1,171 @@
+        assert new_user.check_password(PASSWORD)
+        assert new_user.email == EMAIL
+        new_user.save()  # Profile create just after save user
+        assert new_user.profile.bio == ''
+        assert new_user.profile.profession == ''
+        assert new_user.profile.phone_number == ''
+        assert new_user.profile.city == ''
+        assert new_user.profile.account_type == 'S'
+        assert new_user.profile.meeting_method == 'B'
+
+    # Check if the user was deleted from the database
+    def test_delete_user_from_db(self, new_user):
+        new_user.save()
+        new_user.delete()
+        assert new_user not in User.objects.all()
+        assert new_user.profile not in Profile.objects.all()
+
+    # In this test we delete only the profile part, and check if the profile was deleted and not the whole user
+    def test_delete_only_profile_from_db(self, new_user):
+        new_user.save()
+        new_user.profile.delete()
+        assert new_user in User.objects.all()
+        assert new_user.profile not in Profile.objects.all()
+
+    # Check invalid email
+    def test_validate_email_addr(self):
+        with pytest.raises(ValidationError):
+            user = User(
+                        username=USERNAME,
+                        first_name=FIRSTNAME,
+                        last_name=LASTNAME,
+                        password=PASSWORD,
+                        email="check")
+            user.full_clean()
+
+    # Test for unique username
+    def test_unique_username(self):
+        with pytest.raises(IntegrityError):
+            user1 = User(
+                        username="test",
+                        first_name=FIRSTNAME,
+                        last_name=LASTNAME,
+                        password=PASSWORD,
+                        email=EMAIL)
+            user2 = User(
+                        username="test",
+                        first_name=FIRSTNAME,
+                        last_name=LASTNAME,
+                        password=PASSWORD,
+                        email="test@gmail.com")
+            user1.save()
+            user2.save()
+
+    # Test for invalid profession length
+    def test_invalid_profession_length(self, new_user):
+        with pytest.raises(ValidationError):
+            new_user.save()
+            new_user.profile.profession = LONG_PROFESSION
+            new_user.profile.clean_fields()
+
+    # Check invalid phone number
+    def test_validate_phone_number(self):
+        with pytest.raises(ValidationError):
+            user = User(
+                        username=USERNAME,
+                        first_name=FIRSTNAME,
+                        last_name=LASTNAME,
+                        password=PASSWORD,
+                        email=EMAIL)
+            user.save()
+            user.profile.phone_number = PHONE_NUMBER
+            user.profile.clean_fields()
+
+    # Test for invalid city
+    def test_invalid_city_length(self, new_user):
```

Comment:
Those tests are OK though, while they kind-of testing Django builtins, they can be used to assure a change to the city_length, or registering a test-entity with the wrong length, will be handled.

Reply:
this is the purpose of the function here to check invalid input